torch/distributed/fsdp/sharded_grad_scaler.py

```diff
@@ -285,11 +184,7 @@ def unscale_(self, optimizer: torch.optim.Optimizer) -> None:
         for v in optimizer_state["found_inf_per_device"].values():
             if v.device.type == "cpu":
                 v_on_cuda = v.cuda()
-                future_handles.append(
-                    dist.all_reduce(
-                        v_on_cuda, async_op=True, group=self.process_group
-                    ).get_future()
-                )
+                dist.all_reduce(v_on_cuda, async_op=False, group=self.process_group)
                 v.copy_(v_on_cuda.cpu())
```

Comment:
use async_op=False for now to address buggy `v.copy_(v_on_cuda.cpu())`. will land formal fix https://github.com/pytorch/pytorch/pull/115710 and then rebase on top of it